### ISSUE TITLE
@types/babel__core is an unlisted dependency

### ISSUE THREAD

**Issue by wereHamster** [callstack/linaria#599]
## Environment

- Linaria version: v2.0.0-alpha.2
- Bundler (+ version): babel
- Node.js version: 13
- OS: macOS

## Description

I received the following error from TypeScript:

```
./node_modules/.bin/tsc --build src/tsconfig.json
node_modules/linaria/lib/babel/types.d.ts:94:16 - error TS2665: Invalid module name in augmentation. Module '@babel/core' resolves to an untyped module at '<project>/node_modules/@babel/core/lib/index.js', which cannot be augmented.

94 declare module '@babel/core' {
                  ~~~~~~~~~~~~~
```

The app works fine, it's just the type checke that's not happy. I was able to fix this by installing `@types/babel__core` myself, though it's probably not nice to force users of this library to manually install this dependency.

## Reproducible Demo

https://github.com/timvir/timvir/pull/124

```
npm ci
./node_modules/.bin/tsc --build src/tsconfig.json
```

**Comment by jayu** [comment]
Thanks! I will check it

**Comment by jayu** [comment]
It turns out none of the `@babel` packages had `@types/babel_core` in dependencies. It is installed because it is a dependency of `babel-jest`, that's why there is no error locally when we build linaria before release. `babel-jest` is dev dependency. 
So we should add `@types/babel_core` as a dependency. I don't know how to prevent similar situation in the future

**Comment by wereHamster** [comment]
To be fair, it is really unusual for any npm package to have @types/ in its dependencies… It's only needed when you use TypeScript, and not everyone does.

**Comment by jayu** [comment]
I agree that sucks, but sometimes there is no other solution. We can add it as peerDependency so users will be notified that they may need to install it or add am optional installation step to add it. 
Or the user can build ts with `skipLibCheck` flag. But any of these solutions seems to be good so I would just add it as a dependency.

**Comment by wereHamster** [comment]
@jayu do you know the reason why the changes to the `@babel/core` module have not been pushed to DefinitelyTyped? Are those non-official / private types that are not meant to be used by babel users? If not then we can push those changes upstream and not have to rely on module augmentation.

**Comment by wereHamster** [comment]
Alright, I investigated why the linaria/lib/babel/* .d.ts files are even loaded when I just `import { css } from "linaria"`, and it's because lib/core/css imports ../types, which imports ./babel/utils/loadOptions to get the PluginOptions type. AFAICS PluginOptions does not really use any babel types, so if we could move that to a separate file that would solve that problem.

**Comment by TMaszko** [comment]
@wereHamster Hi, I appreciate that you provided even second PR, but problem is much more complex. I checked when I removed all jest stuff from linaria then your bug occurs with even more errors with types. We are augmenting `@types/babel__core` package so it is a direct dependency which should be listed in linaria's package.json as a `dependency`

**Comment by wereHamster** [comment]
I disagree with your conclusion.

The babel types are not a runtime dependency. The fact that TypeScript picks them up is because babel is exposed as an implementation detail when I `import { css } from "linaria"`. It does not have to be, as my PR demonstrated. For a user of the linaria library, the `css()` function is a pure function, the fact that it is intercepted by the babel macro is immaterial. The fact that you get errors when you remove jest is also irrelevant for users, as tests are only run while working on linaria, not while using it.

**Comment by satya164** [comment]
I think it makes sense to do what you did in the PR @wereHamster. Only change I'd make is to move it to the `react` file, because this type is part of the `styled` API, not the core.

**Comment by TMaszko** [comment]
@wereHamster Thanks for the explanation!

**Comment by wereHamster** [comment]
Moving the type into the react file would mean that when I `import { css } from "linaria"` I would also import the react types. Just something to keep in mind…

**Comment by TMaszko** [comment]
Yeah, So  maybe duplication of that type in both places might be the safest approach.

**Comment by jayu** [comment]
fixed via #620
